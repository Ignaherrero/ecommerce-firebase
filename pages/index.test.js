import React from "react";
import "@testing-library/jest-dom/extend-expect";
import { fireEvent, render } from "@testing-library/react";
import Home from "./index";

describe("<Home/>", () => {
  it("snapshot home", async () => {
    const products = [
      {
        id: 1,
        name: "Product 1",
        price: 100,
      },
    ];

    const { asFragment } = render(<Home products={products} />);
    expect(asFragment()).toMatchSnapshot();
  });

  const mockHandleAddTroley = { handleAddToTroley: jest.fn() };
  it("clicking the button +", () => {
    const products = [
      {
        id: 1,
        name: "Product 1",
        price: 100,
      },
    ];

    const component = render(<Home products={products} />);
    const button = component.getByText("+");
    fireEvent.click(button);
    expect(button).toHaveBeenCalled(1);
  });
});
