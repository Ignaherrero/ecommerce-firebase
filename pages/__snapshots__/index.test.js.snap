// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home/> snapshot home 1`] = `
<DocumentFragment>
  <div
    class="chakra-container css-m7pf44"
  >
    <div
      class="chakra-container css-0"
    >
      <h2
        class="chakra-text css-xi606m"
      >
        tienda tests
      </h2>
      <h3
        class="chakra-text css-0"
      />
      <div
        class="chakra-stack css-wulyol"
      >
        <div
          class="css-1a09ufk"
        >
          <p
            class="chakra-text css-0"
          />
          <p
            class="chakra-text css-xnoh5a"
          >
            $ 100
          </p>
          <div
            class="css-1l4w6pd"
          >
            <img
              alt="product"
              class="chakra-image css-v7v99c"
            />
          </div>
          <div
            class="chakra-button__group css-1ind3v2"
            role="group"
          >
            <button
              class="chakra-button css-2qv00o"
              type="button"
            >
              +
            </button>
            <button
              class="chakra-button css-2qv00o"
              type="button"
            >
              -
            </button>
          </div>
        </div>
      </div>
      <p
        class="chakra-text css-0"
      >
        Total: $0
      </p>
      <button
        class="chakra-button css-5c2176"
        disabled=""
        type="button"
        variantcolor="blue"
      >
        Vaciar carrito
      </button>
    </div>
  </div>
</DocumentFragment>
`;
